docs/BRANCHING.md

```diff
@@ -0,0 +1,51 @@
+# Branching Strategy for releases
+
+## General Considerations
+
+- Generally the master branch is the one where current development is happening.
+- One objective would be to arrive to a point where the master branch can be released very often (every day?).
+  This means the master branch is releasable at all times. I believe this strategy is called **Continuous Delivery**.
+- At this point we don't think it's possible to do that until the testing infrastructure has been improved.
+- Nevertheless, the branching strategy takes this objective into account.
+- To further support this objective, the strategy wil try to minimize the number of long-lasting branches.
+
+## Release numbering
+- example 4.3.0
+- The first number (4 in this example) is incremented for major changes in the software. What major change mean has to be decided.
```

Comment:
Major = impacts the rules, either errors or warnings.

Reply:
So do you mean adding a rule for example will require a major number change?

Reply:
Correct - any additions or modifications to validation rules.

Reply:
How often are the rules changed?

Reply:
I would expect at the once every 3 month cadence we talked about (we're talking about modifying equal_shape_dist_traveled_diff_coordinates and trip_distance_exceeds_shape_distance this quarter, for instance, and David's [expired_calendar PR](https://github.com/MobilityData/gtfs-validator/pull/1646) would be a rule modification). We also need to add new fares rules.

Reply:
So that means we would never really have minor releases, if there are changes of rules that often. Can we qualify the changes in the rules to make some more important (and impactful) than others?

Reply:
@jcpitre I think the reaction would just be to release more :) For example, we could do a minor release for everything we've merged this past month. (Not suggesting it, but going forward after this quarter)